rfcs/2021-11-12-9811-vrl-vm.md

```diff
@@ -0,0 +1,164 @@
+# RFC 9811 - 2021-11-12 - VRL bytecode VM
+
+This RFC proposes implementing a bytecode VM. VRL will be compiled to the bytecode
+and executed by this VM, with the aim to significantly improve the performance of
+executing VRL scripts.
+
+## Context
+
+This RFC is a follow on from
+[rfcs/2021-10-14-9811-vrl-performance.md](https://github.com/vectordotdev/vector/pull/9812)
+
+## Scope
+
+### In scope
+
+This RFC is purely about developing a bytecode VM for VRL.
```

Comment:
I'd like this RFC to be more explicit about the goals around making this as safe and easy to use. That includes:

1. Documentation/code comments
2. Testing (unit/fuzzing/...)
3. Preventing misuse through a safe API